noir-projects/aztec-nr/aztec/src/event/event_interface.nr

```diff
@@ -1,9 +1,10 @@
 use crate::context::PrivateContext;
 use crate::note::note_header::NoteHeader;
-use dep::protocol_types::{grumpkin_point::GrumpkinPoint, abis::function_selector::FunctionSelector};
+use dep::protocol_types::{grumpkin_point::GrumpkinPoint, abis::event_selector::EventSelector};
 
-trait EventInterface<N> {
-    // Should be autogenerated by the #[aztec(event)] macro unless it is overridden by a custom implementation
-    fn _selector(self) -> FunctionSelector;
-    fn to_be_bytes(self, randomness: Field) -> [u8; N];
+trait EventInterface<M> {
+    fn private_to_be_bytes(self, randomness: Field) -> [u8; M];
+    // More than one generic breaks this
+    // fn to_be_bytes(self) -> [u8; M];
```

Comment:
Don't this not work fine with `M, N` as generics? Not sure exactly how things will be working in public, maybe @fcarreiro have some insights?

Reply:
Looks like transpiler errors are popping up every now and then. In all cases so far it seems to be stale JSON files and deleting target and rebuilding contracts seems to solve it. If you find the root cause LMK. I'll try to look on my side.